service/src/test/java/bio/terra/drshub/services/DrsApiFactoryTest.java

```diff
@@ -1,57 +1,80 @@
 package bio.terra.drshub.services;
 
-import static org.mockito.ArgumentMatchers.anyString;
-import static org.mockito.Mockito.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
 
-import bio.terra.drshub.BaseTest;
 import bio.terra.drshub.config.DrsProvider;
 import bio.terra.drshub.config.MTlsConfig;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Tag;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.web.util.UriComponents;
 import org.springframework.web.util.UriComponentsBuilder;
 
 @Tag("Unit")
-public class DrsApiFactoryTest extends BaseTest {
+@ExtendWith(MockitoExtension.class)
+class DrsApiFactoryTest {
 
-  @Autowired DrsApiFactory drsApiFactory;
+  @Spy private RestTemplateFactory restTemplateFactory;
```

Comment:
In general I think we should avoid writing tests that use `spy()`, as it makes it harder to reason about what is being tested vs what is being mocked. In this case, why is `Spy` needed for `RestTemplateFactory`?

Reply:
I also don't like using spies in tests.

I initially removed it in favor of a mock, but added it back in following a screech from [Codecov](https://github.com/DataBiosphere/terra-drs-hub/runs/19306586354) that the testing coverage had gone down significantly as a result: https://github.com/DataBiosphere/terra-drs-hub/pull/92/commits/4b491e82bcce88d9d3151c88c68f559a4aadaec7

I could split this out into three layers instead of two (existing is just `DrsApiFactory`), but I got stuck when trying to test `RestTemplateFactory` in its own unit test suite to maintain coverage.  I was unsure what to verify: on creation of these RestTemplates, they end up locking down most of their contents.  Any ideas?

Reply:
Thanks for your help -- resolved in [83b86af](https://github.com/DataBiosphere/terra-drs-hub/pull/92/commits/83b86af48313d69a3ad8e25a02606245620b379d)!